python/plugins/processing/algs/gdal/GdalAlgorithm.py

```diff
@@ -115,6 +116,11 @@ def getOgrCompatibleSource(self, parameter_name, parameters, context, feedback,
                 #for the command line preview since it has no meaning outside of a QGIS session!
                 ogr_data_path = GdalUtils.ogrConnectionStringAndFormatFromLayer(input_layer)[0]
                 ogr_layer_name = GdalUtils.ogrLayerName(input_layer.dataProvider().dataSourceUri())
+        elif input_layer.dataProvider().name().lower() == 'wfs':
```

Comment:
Could you add a unit test covering this?

Reply:
As I wrote in the previous comment, I will be happy to cover it with test. But need to find fast and reliable public WFS server (to create a valid layer). Already tried to create layer using fake connection string, but as resulting layer is not valid calls to its `dataProvider()` method returns `None`. Or I miss something and it can be done without layer creation?

Reply:
You could try QgsMapLayer.providerType(). It's badly named, but this will return the data provider name even when the layer is invalid/provider doesn't exist.

Reply:
Unfortunately this does not work. While provider recognized correctly, layer still invalid and can not be added to project.

Reply:
Ok, let's merge anyway?

Reply:
I would say yes, if this looks good.